src/pages/trainer/training/edit/actions/updateEditor.ts

```diff
@@ -0,0 +1,15 @@
+import {trainerActionEnums} from '../../../../../common/actionEnums/trainer';
+import {textAreaTool} from '../../../../../common/ui/tools/textAreaTool';
+
+export const updateEditorAction = (editor: HTMLTextAreaElement, caret: string, offset: number) => {
+  const contentWithCaret = textAreaTool.insertAtCaretGetText(editor, caret, offset);
+  const cursorStartPosition = textAreaTool.caculateStartCursorPositionPlusOffset(editor, offset);
```

Comment:
No, textAreaTool os fine enclosed un that class.

It's accessing a Dom elementos from the ActionScript what sounds strange to me, I think It should fallo into component responsiblity